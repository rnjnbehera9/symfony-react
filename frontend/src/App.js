import './App.css';
import React from 'react';
import MovieList from './MovieList';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';

function App() {
    return (
        <Router>
            <Routes>
                <Route path="/" element={<MovieList/>} />
            </Routes>
        </Router>
    );
}

export default App;
