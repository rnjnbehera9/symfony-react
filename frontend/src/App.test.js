import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders MovieList component when navigating to /', () => {
  render(
      <App initialEntries={['/']}/>
  );

  expect(screen.getByText(/My Movies/i)).toBeInTheDocument();
});
it('renders correctly', () => {
  const tree = renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

