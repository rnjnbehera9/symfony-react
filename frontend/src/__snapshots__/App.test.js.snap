// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="ui header"
    >
      My Movies
    </h1>
    <div>
      <table
        className="ui table"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Title
            </th>
            <th>
              Movie Count
            </th>
            <th>
              Actions
            </th>
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
      <div>
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <label>
              Title: 
            </label>
            <input
              onChange={[Function]}
              placeholder="enter movie title"
              required={true}
              value={Array []}
            />
          </div>
          <button
            className="ui button"
            onClick={[Function]}
            type="submit"
          >
            Add Movie
          </button>
        </form>
      </div>
    </div>
  </header>
</div>
`;
